### PR TITLE
Add STUN Support

### PR DESCRIPTION
**Pull request by Oborichkin**
- Added basic support for `Binding` method of STUN ([RFC 8489](https://www.rfc-editor.org/rfc/rfc8489.html#section-14.7)).
- Added attributes:
  - `USERNAME`
  - `MESSAGE-INTEGRITY`
  - `XOR-MAPPED-ADDRESS`
  - `USE-CANDIDATE`
  - `PRIORITY`
  - `FINGERPRINT`
  - `ICE-CONTROLLING`
  - `GOOG-NETWORK-INFO`

### CODE REVIEW

**Comment by p-l-** [comment]
Hi @Oborichkin!
Could you please rebase against current master for the CI tests?

File: scapy/contrib/stun.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        XShortField("type", 0x0000),
+        FieldLenField("length", None, length_of="value"),
+        PadField(StrLenField("value", "", length_from=lambda pkt:pkt.length), align=4)
+    ]
+
+    @classmethod
+    def dispatch_hook(cls, _pkt=None, *args, **kwargs):
+        if _pkt and len(_pkt) >= 2:
+            t = struct.unpack("!H", _pkt[:2])[0]
+            return _stun_tlv_class.get(t, "STUNGenericTlv")
```
Comment (p-l-) [inline scapy/contrib/stun.py:85]:
```suggestion
            return _stun_tlv_class.get(t, cls)
```

**Comment by Oborichkin** [comment]
Hi @p-l- !

> Could you please rebase against current master for the CI tests?

done!

File: scapy/contrib/stun.py (hunk 1 above)
Comment (Oborichkin) [reply scapy/contrib/stun.py]: done

**Comment by p-l-** [comment]
This PR seems to break some of the tests (apparently Python 2). Could you have a look?

**Comment by jkldgoefgkljefogeg** [comment]
Can you make it RFC8489 compliant?

**Comment by Oborichkin** [comment]
> This PR seems to break some of the tests (apparently Python 2). Could you have a look?

Yep, done

**Comment by Oborichkin** [comment]
> Can you make it RFC8489 compliant?

Oops, wrote a wrong RFC number in PR. It is RFC 8489 compliant

**Review by p-l- — APPROVED** [review]

**Comment by evverx** [comment]
@Oborichkin it's probably a long shot but I thought I'd ask anyway. I opened https://github.com/secdev/scapy/pull/3877 to fix  https://github.com/secdev/scapy/issues/3876 and that PR broke the STUN tests. I wonder why the tests expect for example
```sh
0x77e7f03e7f9e1996be42339159db1f682147bcfc
```
instead of
```sh
0xfcbc4721681fdb59913342be96199e7f3ef0e777
```
when `raw` contains
```sh
\xfc\xbc\x47\x21\x68\x1f\xdb\x59\x91\x33\x42\xbe\x96\x19\x9e\x7f\x3e\xf0\xe7\x77
```
given that `STUNMessageIntegrity(hmac_sha1=0x77e7f03e7f9e1996be42339159db1f682147bcfc)` produces those bytes in reverse order:
```
>>> hexdump(STUNMessageIntegrity(hmac_sha1=0x77e7f03e7f9e1996be42339159db1f682147bcfc))
0000  00 08 00 14 77 E7 F0 3E 7F 9E 19 96 BE 42 33 91  ....w..>.....B3.
0010  59 DB 1F 68 21 47 BC FC                          Y..h!G..
```
? My guess would be that it was copied from `STUN(raw).attributes` (which wasn't correct at the time due to that bug) but it could be that I'm missing something and the STUN module actually somehow relies on that bug.

**Comment by oborichkin** [comment]
@evverx It's very unlikely that STUN relies on that bug :)
At the time of writing STUN dissectors I was unaware of that issue, so I thought it was an expected behavior. Now comparing output with wireshark it's obvious - test should be fixed

**Pull request merged**